### PR TITLE
feat(smolvla): add pretrained weight loading from lerobot/smolvla_base

### PR DESCRIPTION
**Pull request by Daankrol**
> Keeping in draft mode until benchmarks are done.

## Summary

Add support for loading full pretrained SmolVLA checkpoint from HuggingFace (`lerobot/smolvla_base`), following the same pattern already implemented for Pi05 (`Pi05(pretrained_name_or_path="lerobot/pi05_base")`).

**Problem:** Physical-ai-studio had no mechanism to initialize SmolVLA from the pretrained `lerobot/smolvla_base` checkpoint containing the VLM backbone + action expert + all paper-trained weights. Without this, fine-tuning starts from a randomly initialized action expert, which is one of the key factors causing poor performance on SO-101 robots (actuators wiggle with no real movement).

**Solution:** `SmolVLA(pretrained_name_or_path="lerobot/smolvla_base")` now:
- Downloads `config.json`, `model.safetensors`, and normalizer stats from HF
- Architecture params (`prefix_length`, `num_expert_layers`, `pad_language_to`, etc.) come from the pretrained config.json; only training-time settings (optimizer, scheduler, finetuning flags) are overridable by the caller
- `load_vlm_weights` is forced to `False` to avoid re-downloading the VLM backbone already included in the full checkpoint
- State dict keys are remapped (`model.` → `_model.`) to match SmolVLAModel's internal layout vs lerobot's SmolVLAPolicy layout
- During fine-tuning, `setup()` replaces pretrained normalization stats with the actual training dataset's stats

### CODE REVIEW

**Comment by sovrasov** [comment]
Thanks! I've completed some experiments to establish a baseline LIBERO performance (like @eugene123tw did for pi05)
Here's lerobot v0.4.4 benchmark command:
```bash
lerobot-eval \
  --policy.path="HuggingFaceVLA/smolvla_libero" \
  --env.type=libero \
  --env.task=libero_10 \
  --eval.batch_size=1 \
  --eval.n_episodes=10 \
  --env.max_parallel_tasks=1 \
  --env.episode_length=500 \
```
Baseline results:
```
"libero_10": {
      "avg_sum_reward": 0.45,
      "avg_max_reward": 0.45,
      "pc_success": 45.0,
      "n_episodes": 100,
}
```
I have also per task results and videos if needed.
Ideally, in this PR we should achieve a similar result with:

```python
from physicalai.policies import SmolVLA
from physicalai.benchmark import LiberoBenchmark

from libero.libero import get_libero_path

if __name__ == "__main__":
    # Load from HuggingFace (equivalent to your lerobot command)
    policy = SmolVLA(
        pretrained_name_or_path="HuggingFaceVLA/smolvla_libero",
    )

    policy.to("cuda")

    # Evaluate on LIBERO (equivalent to your lerobot eval)
    benchmark = LiberoBenchmark(
        task_suite="libero_10",
        video_dir="videos/smolvla_libero",
        record_mode="failures",
    )
    results = benchmark.evaluate(policy)
    print(results.summary())
```
